### PR TITLE
Unicam FS/FE timing GPIO

### PR DESCRIPTION
**Pull request by 6by9**
Replicates the firmware option to enable a GPIO to follow CSI frame start/end timings.

https://forums.raspberrypi.com/viewtopic.php?t=368437

### CODE REVIEW

File: arch/arm/boot/dts/broadcom/bcm270x-rpi.dtsi [hunk 1]
```diff
@@ -97,6 +97,19 @@
 		drm_fb0_vc4 = <&aliases>, "drm-fb0=",&vc4;
 		drm_fb1_vc4 = <&aliases>, "drm-fb1=",&vc4;
 		drm_fb2_vc4 = <&aliases>, "drm-fb2=",&vc4;
+
+		cam1_sync = <&csi1>, "sync-gpios:0=", <&gpio>,
+			    <&csi1>, "sync-gpios:4",
+			    <&csi1>, "sync-gpios:8=0"; /* GPIO_ACTIVE_HIGH */
```
Comment (pelwell) [inline arch/arm/boot/dts/broadcom/bcm270x-rpi.dtsi]: This is OK, but you can use the named constant:
```
		cam1_sync = <&csi1>, "sync-gpios:0=", <&gpio>,
			    <&csi1>, "sync-gpios:4",
			    <&csi1>, "sync-gpios:8=", <GPIO_ACTIVE_HIGH>;
```

**Comment by 6by9** [comment]
@naushir What's your view on this? It could be done from userspace, but with a large round trip delay.

**Comment by naushir** [comment]
Looks reasonable to me.  Probably want to replicate this in the cfe driver?

**Comment by 6by9** [comment]
> Looks reasonable to me. Probably want to replicate this in the cfe driver?

Probably yes (caveat around multiple virtual channels), but I'll leave that up to you :-)

The OP was really after the simple flash driver function from the legacy stack, but I don't believe flash has really been considered in libcamera as yet.

As this can be merged without impacting the normal flow of things, I'll drop the draft status even though the OP hasn't come back on it.

**Comment by 6by9** [comment]
Any objections to merging this then?

**Comment by naushir** [comment]
LGTM

**Pull request merged**